components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/handlers/DefaultKeyValidationHandler.java

```diff
@@ -170,32 +179,69 @@ public boolean validateScopes(TokenValidationContext validationContext) throws A
                 + ":" +
                 validationContext.getHttpVerb();
 
-        boolean isValid = false;
+        Set<OAuth2ScopeValidator> oAuth2ScopeValidators = OAuthServerConfiguration.getInstance()
+                .getOAuth2ScopeValidators();
+        //validate scope for filtered validators from db
+        String[] scopeValidators;
+        OAuthAppDO appInfo;
+        OAuthAppDAO oAuthAppDAO = new OAuthAppDAO();
+
         try {
-            Set<OAuth2ScopeValidator> oAuth2ScopeValidators = OAuthServerConfiguration.getInstance()
-                    .getOAuth2ScopeValidators();
-            //validate scope from all the validators
+            appInfo = oAuthAppDAO.getAppInformation(clientId);
+            scopeValidators = appInfo.getScopeValidators();     //get scope validators from the DB
+            ArrayList<String> appScopeValidators = new ArrayList<>(Arrays.asList(scopeValidators));
+
+            if (ArrayUtils.isEmpty(scopeValidators)) {
```

Comment:
What happen to existing application which doens't have scope validators and we should support the default behavior.

Reply:
When the applications haven't been configured with any scope validators, it should return true as the default behavior.